### PR TITLE
Update help docs for new design and scores

### PR DESCRIPTION
**Pull request by mit-mit**

### CODE REVIEW

File: doc/help.md [hunk 1]
```diff
@@ -1,70 +1,205 @@
...
+Default listings use a composite of the three scores to sort packages. The
+ranking is based on the composite score, and if applicable, the platform
+specificity and the text match score of the search team.
+
+# Publishing packages
+
+[Pub][pub-tool] isn't just for using other people's packages.
+It also allows you to share your packages with the world.
+If you have a useful project and you want others to be able to use it,
+use the [`dart pub publish`][pub-publish] command.
```
Comment (isoos) [inline doc/help.md:53]: `pub publish`

File: doc/help.md [hunk 2]
```diff
@@ -84,48 +219,35 @@ include: package:pedantic/analysis_options.yaml
...
-*How can you improve your maintenance score?*
+  - Works with the latest stable Dart SDK.
+  - Works with the latest stable Flutter SDK (if applicable).
+  - Works with the latest versions of all pendendencies.
 
-Click your package's overall score to see the Analysis page, which has
-suggestions for improving the package's score. Fix them, and release
-at least one new version every year to keep your maintenance score up.
+To determine if your package supports the latest versions of dependencies prior
+to publishing, run the `dart pub outdated` command (Dart SDK) or `flutter pub
```
Comment (isoos) [inline doc/help.md:231]: `pub outdated`

File: doc/help.md [hunk 3]
```diff
@@ -84,48 +219,35 @@ include: package:pedantic/analysis_options.yaml
...
+You can view your *pub points report* to get suggestions before publishing, by
+running `pana` locally:
 
-You can find the overall score either near the top of the package's page
-or to the right of your package in any listing on this site.
+  1. When run `pana` will make modifications to the package, so start by making
+     a copy of the directory holding your package: `cp ~/dev/mypkg ~/tmp/mypkg`
 
-## Ranking
+  1. Make sure you have the latest pana tool: `dart pub global activate pana` (`pana`
```
Comment (isoos) [inline doc/help.md:246]: `pub global activate pana`

File: doc/help.md [hunk 4]
```diff
@@ -84,48 +219,35 @@ include: package:pedantic/analysis_options.yaml
...
+     changes frequently, so run this again frequenrly to update the pana tool)
 
-Default listings use composite scoring to sort packages. The score is
-based on the overall score, and if applicable, the platform specificity
-and the text match score is also factored in.
+  1. Run pana on the copy we made earlier: `dart pub global run pana --source path ~/tmp/mypkg`
```
Comment (isoos) [inline doc/help.md:249]: `pub global run pana ~/tmp/mypkg` (I think we no longer require `--source path` for local testing).

File: doc/help.md [hunk 5]
```diff
@@ -84,48 +219,35 @@ include: package:pedantic/analysis_options.yaml
...
+running `pana` locally:
 
-You can find the overall score either near the top of the package's page
-or to the right of your package in any listing on this site.
+  1. When run `pana` will make modifications to the package, so start by making
+     a copy of the directory holding your package: `cp ~/dev/mypkg ~/tmp/mypkg`
 
-## Ranking
+  1. Make sure you have the latest pana tool: `dart pub global activate pana` (`pana`
+     changes frequently, so run this again frequenrly to update the pana tool)
```
Comment (isoos) [inline doc/help.md:247]: s/frequenrly/frequently/

File: doc/help.md [hunk 6]
```diff
@@ -1,70 +1,205 @@
...
+The search UI also supports filters:
+
+  - On the left of the search bar, select 'Dart' or 'Flutter' depending on the
+    SDK you are developing with.
+
+  - On the right of the search bar, click the platform filter buttons to filter
+    results to packages supporting the platforms you need support for.
+
+  - To use platform filters for non-stable platforms (e.g. Flutter Desktop), use
+    the Advanced option on the far right of the search bar.
```
Comment (jonasfj) [inline doc/help.md:40]:
```suggestion
  - To use platform filters for prerelease platforms (e.g. Flutter Desktop), use
    the Advanced option on the far right of the search bar.
```

File: doc/help.md [hunk 7]
```diff
@@ -1,70 +1,205 @@
...
+### Follow Dart file conventions
+
+Dart packages are expected to follow certain file conventions for how to
+[organize a package](https://dart.dev/guides/libraries/create-library-packages#organizing-a-library-package.
+Most importantly make sure to:
+
+  - Provide a [`pubspec.yaml` file](https://dart.dev/tools/pub/pubspec). Ensure
+    all Urls are valid and use a secure `https:` scheme.
+
+  - Provide a [](https://dart.dev/tools/pub/package-layout#license) file using
```
Comment (jonasfj) [inline doc/help.md:140]:
```suggestion
  - Provide a [`LICENSE`](https://dart.dev/tools/pub/package-layout#license) file using
```

File: doc/help.md [hunk 8]
```diff
@@ -1,70 +1,205 @@
...
+    an an [OSI-approved license](https://opensource.org/licenses).
```
Comment (jonasfj) [inline doc/help.md:141]: We do not require OSI-approved licenses. I think encouraging it would be great, but afaik we have no scoring on this.

Comment (jonasfj) [reply doc/help.md:141]: Only scoring on the fact that a license is present.

File: doc/help.md (hunk 1 above)
Comment (mit-mit) [reply doc/help.md:53]: Done

File: doc/help.md (hunk 2 above)
Comment (mit-mit) [reply doc/help.md:231]: Done

File: doc/help.md (hunk 4 above)
Comment (mit-mit) [reply doc/help.md:249]: Done

File: doc/help.md (hunk 5 above)
Comment (mit-mit) [reply doc/help.md:247]: Done

File: doc/help.md (hunk 6 above)
Comment (mit-mit) [reply doc/help.md:40]: I don't think Flutter uses that terminology. Changed to alpha platforms.

File: doc/help.md (hunk 7 above)
Comment (mit-mit) [reply doc/help.md:140]: Done

File: doc/help.md (hunk 8 above)
Comment (mit-mit) [reply doc/help.md:141]: Added preferably

**Pull request merged**